toxav/ring_buffer.h

```diff
@@ -0,0 +1,18 @@
+#ifndef RING_BUFFER_H
```

Comment:
I don't know the point of RingBuffer. This PR just moves it out of toxcore where it doesn't belong. @mannol can add documentation.